### PR TITLE
add multiSig for whitelisted officers to approve the bid

### PR DESCRIPTION
**Pull request by michaelwwn**
Key Modifications and Additions
**Whitelist Management:** Added functions addToWhitelist and removeFromWhitelist to manage which officers are allowed to approve or reject bids, with corresponding events for tracking these actions.
**Multi-Signature Approval: I**ntroduced ApprovalData struct to track the count of approvals for each bid and which officers have approved it. The approveBid and rejectBid functions now require a minimum of two approvals before changing the bid's status, ensuring no single officer can unilaterally approve or reject a bid.
**Status Flags:** Added boolean flags isApproved and isRejected within ApprovalData to prevent a bid from being approved and then rejected (or vice versa) after reaching the minimum approval count.
Modifiers: Implemented notAlreadyApproved to prevent an officer from

### CODE REVIEW

**Review by emilio-quintana-dev — APPROVED** [review]
:shipit:

File: contracts/ApprovalWorkflow.sol [hunk 1]
```diff
@@ -1,53 +1,102 @@
...
-        });
+    modifier notAlreadyApproved(uint256 _rfpId) {
+        require(!approvalData[_rfpId].approvers[msg.sender], "Officer has already approved this bid");
+        _;
+    }
+
+    function addToWhitelist(address _officer) public onlyOwner {
+        whitelist[_officer] = true;
+        emit OfficerAddedToWhitelist(_officer);
+    }
```
Comment (JeremyHarithLim) [inline contracts/ApprovalWorkflow.sol]: - Should there be a check whether the officer address already exists within the whitelist? Also are we going to allow other officers to add new officers? 
_I vaguely recall that that there exists Access Control within OpenZeppelin, do we want to consider using that for more granular control of officers?_ 
- Also do we want to allow certain specific officers to add other officers or just the owner of the contract?

File: contracts/ApprovalWorkflow.sol [hunk 2]
```diff
@@ -1,53 +1,102 @@
...
 
-        emit BidApproved(_rfpId, _vendor);
+    function removeFromWhitelist(address _officer) public onlyOwner {
+        whitelist[_officer] = false;
+        emit OfficerRemovedFromWhitelist(_officer);
```
Comment (JeremyHarithLim) [inline contracts/ApprovalWorkflow.sol]: - Same here as above, do we want to use OpenZeppelins's Acccess Control? Although I do think we might not have enough time so this should suffice.

**Review by JeremyHarithLim — CHANGES_REQUESTED** [review]

File: contracts/ApprovalWorkflow.sol (hunk 1 above)
Comment (michaelwwn) [reply contracts/ApprovalWorkflow.sol]: changed to Openzeppelin Contract

File: contracts/ApprovalWorkflow.sol (hunk 2 above)
Comment (michaelwwn) [reply contracts/ApprovalWorkflow.sol]: changed to Openzeppelin Contract

**Review by JeremyHarithLim — APPROVED** [review]

**Pull request merged**